File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.go
func twoSum(numbers []int, target int) []int {
    i := 0; j := len(numbers)-1
    for i<j{
        if numbers[i]+numbers[j] == target{
            break
        }
        if numbers[i]+numbers[j] > target{
            j--
        }else{
            i++
        }
    }

    if i == j{
        return []int{-1, -1}
    }
    return []int{i+1, j+1}
}

File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] soln = new int[]{-1, -1};
        Map<Integer, Integer> map = new HashMap<>();
        for(int i=0; i<nums.length; i++){
            int comp = target - nums[i];
            if (map.get(comp) == null){
                map.put(nums[i], i);
            }else{
                soln[0] = map.get(comp);
                soln[1] = i;
                return soln;
            }
        }
        return soln;
    }
}
